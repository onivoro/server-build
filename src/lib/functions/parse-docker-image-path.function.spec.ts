import { parseDockerImagePath } from './parse-docker-image-path.function';

describe('parseDockerImagePath', () => {
  describe('GIVEN the tag is present', () => {
    it.each([
      ['314218077608.dkr.ecr.us-east-2.amazonaws.com/evolve:api-account',],
    ])('returns repo, tag, and the combination thereof', (input) => {
      expect(parseDockerImagePath(input)).toMatchSnapshot();
    });
  });

  describe('GIVEN the tag is absent', () => {
    it.each([
      ['314218077608.dkr.ecr.us-east-2.amazonaws.com/evolve',],
    ])('returns repo', (input) => {
      expect(parseDockerImagePath(input)).toMatchSnapshot();
    });
  });
});