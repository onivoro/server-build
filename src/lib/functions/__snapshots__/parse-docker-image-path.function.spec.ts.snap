// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`parseDockerImagePath GIVEN the tag is present returns repo, tag, and the combination thereof 1`] = `
{
  "repo": "314218077608.dkr.ecr.us-east-2.amazonaws.com/evolve",
  "repoColonTag": "314218077608.dkr.ecr.us-east-2.amazonaws.com/evolve:api-account",
  "tag": "api-account",
}
`;

exports[`parseDockerImagePath GIVEN the tag is absent returns repo 1`] = `
{
  "repo": "314218077608.dkr.ecr.us-east-2.amazonaws.com/evolve",
  "repoColonTag": undefined,
  "tag": undefined,
}
`;
